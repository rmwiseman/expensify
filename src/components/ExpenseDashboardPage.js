import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFitlers from './ExpenseListFilters';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;