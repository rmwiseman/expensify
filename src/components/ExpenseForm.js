import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

//const now = moment();
//console.log(now.format('D MMM YYYY'));

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: moment(props.expense ? props.expense.createdAt : undefined),
      calendarFocused: false,
      error: undefined
    };
  }
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  amountPtrn = /^\d+(?:\.\d{0,2})?$/;
  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || this.amountPtrn.test(amount)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = (createdAt) => {
    createdAt && this.setState(() => ({ createdAt }));
  };
  onCalendarFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({ error: 'Please provide a description and an amount.' }));
    } else {
      this.setState(() => ({ error: undefined }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          className="text-input"
          type="text"
          placeholder="Description"
          autoFocus
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          className="text-input"
          type="text"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onCalendarFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          className="textarea"
          placeholder="Add a note for expense (optional)"
          value={this.state.note}
          onChange={this.onNoteChange}
        >
        </textarea>
        <div>
          <button className="button">{this.props.expense ? 'Save Changes' : 'Add Expense'}</button>
        </div>
      </form>
    );
  }
}