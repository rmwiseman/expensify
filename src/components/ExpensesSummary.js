import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import totalExpenses from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
  <h1>Showing {expenseCount} expense{expenseCount == 1 || 's'} totalling &pound;{
    numeral(expensesTotal/100).format('0,0.00')
  }</h1>
  </div>
);

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: totalExpenses(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary); // weird API!