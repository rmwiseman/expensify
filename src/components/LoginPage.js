import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/authentication';

export const LoginPage = ({ startLogin }) => (
  <div>
    <h1>Expensify Login</h1>
    <button onClick={startLogin} >Log In</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);