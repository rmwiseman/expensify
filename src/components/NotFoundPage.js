import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    Nope, not found.  Why not <Link to="/">go home</Link> instead?
  </div>
);

export default NotFoundPage;