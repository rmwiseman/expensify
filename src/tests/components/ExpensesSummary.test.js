import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123.45} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={78} expensesTotal={1234.56} />);
  expect(wrapper).toMatchSnapshot();
});