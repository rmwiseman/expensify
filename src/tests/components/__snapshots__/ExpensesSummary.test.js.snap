// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Showing 
    1
     expense
     totalling £
    1.23
  </h1>
</div>
`;

exports[`should correctly render expenses summary with one expense 2`] = `
<div>
  <h1>
    Showing 
    78
     expense
    s
     totalling £
    12.35
  </h1>
</div>
`;
